server/tilesource/tiff.py

```diff
@@ -56,49 +56,46 @@ def __init__(self, item, **kwargs):
         largeImagePath = self._getLargeImagePath()
         lastException = None
 
-        directories = {}
-        first = None
+        alldir = []
         for directoryNum in itertools.count():
             try:
                 td = TiledTiffDirectory(largeImagePath, directoryNum)
             except ValidationTiffException as lastException:
                 continue
             except TiffException as lastException:
                 break
-            # Ensure all directories have the same tile size
-            if first:
-                if (td.tileWidth != first.tileWidth or
-                        td.tileHeight != first.tileHeight):
-                    continue
-            else:
-                first = td
+            if not td.tileWidth or not td.tileHeight:
+                continue
             level = int(math.ceil(math.log(max(
                 float(td.imageWidth) / td.tileWidth,
                 float(td.imageHeight) / td.tileHeight)) / math.log(2)))
-            # If two directories are at the same level, keep the higher
-            # resolution one
-            if (level in directories and td.imageWidth * td.imageHeight <
-                    directories[level].imageWidth *
-                    directories[level].imageHeight):
-                continue
-            directories[level] = td
-
-        if not len(directories):
+            alldir.append((td.tileWidth * td.tileHeight, level, td))
+        if not len(alldir):
             msg = 'File %s didn\'t meet requirements for tile source: %s' % (
                 largeImagePath, lastException)
             logger.debug(msg)
             raise TileSourceException(msg)
+        alldir.sort()
```

Comment:
A comment about what this loop is doing, and all the negative indexing, would be helpful for future developers.